vector/src/main/java/im/vector/riotx/core/platform/VectorBaseActivity.kt

```diff
@@ -180,6 +185,34 @@ abstract class VectorBaseActivity : AppCompatActivity(), HasScreenInjector {
         }
     }
 
+    private fun handleGlobalError(globalError: GlobalError) {
+        when (globalError) {
+            is GlobalError.InvalidToken         ->
+                handleInvalidToken(globalError)
+            is GlobalError.ConsentNotGivenError ->
+                consentNotGivenHelper.displayDialog(globalError.consentUri,
+                        activeSessionHolder.getActiveSession().sessionParams.homeServerConnectionConfig.homeServerUri.host ?: "")
+        }
+    }
+
+    protected open fun handleInvalidToken(globalError: GlobalError.InvalidToken) {
+        Timber.w("Invalid token event received")
+        if (mainActivityStarted) {
+            return
+        }
+
+        mainActivityStarted = true
+
+        MainActivity.restartApp(this,
```

Comment:
restart is making me thinking that there still is the exit(0) thing

Reply:
No exit(0) :)